enterprise/tasks.py

```diff
@@ -182,3 +207,130 @@ def send_sso_configured_email(
         )
         LOGGER.exception(message)
         raise exc
+
+
+@shared_task
+@set_code_owner_attribute
+def send_group_membership_invitation_notification(
+    enterprise_customer_uuid,
+    membership_uuids,
+    act_by_date,
+    catalog_uuid
+):
+    """
+    Send braze email notification when member is invited to a group.
+
+    Arguments:
+        * enterprise_customer_uuid (string)
+        * memberships (list)
+        * act_by_date (datetime)
+        * catalog_uuid (string)
+    """
+    enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+    braze_client_instance = BrazeAPIClient()
+    enterprise_catalog_client = EnterpriseCatalogApiClient()
+    braze_trigger_properties = {}
+    contact_email = enterprise_customer.contact_email
+    enterprise_customer_name = enterprise_customer.name
+    braze_trigger_properties['contact_admin_link'] = braze_client_instance.generate_mailto_link(contact_email)
+    braze_trigger_properties['enterprise_customer_name'] = enterprise_customer_name
+    braze_trigger_properties[
+        'catalog_content_count'
+    ] = enterprise_catalog_client.get_catalog_content_count(catalog_uuid)
+
+    braze_trigger_properties['act_by_date'] = act_by_date.strftime('%B %d, %Y')
+    pecu_emails = []
+    ecus = []
+    for membership_uuid in membership_uuids:
+        group_membership = enterprise_group_membership_model().objects.filter(
+            uuid=membership_uuid
+        )
```

Comment:
This is going to perform one query per memberhsip, but you could actually do this in a single query like so:
```python
membership_records = enterprise_group_membership_model().objects.filter(uuid__in=membership_uuids)
for group_membership in membership_records:
    if group_membership.pending_enterprise_customer_user is not None:
        pecu_emails.append(group_membership.pending_enterprise_customer_user.user_email)
    else:
        ecus.append({
            'user_email': group_membership.enterprise_customer_user.user_email,
            'user_id': group_membership.enterprise_customer_user.user_id
        })
```